### DISCUSSION TITLE
Support Japanese vertical writing supported by EPUB3 orz

### DISCUSSION THREAD

**Discussion by mark10als** [sumatrapdfreader/sumatrapdf#3277] [Ideas]
I also thought that this request should be passed on to the MuPDF project.
I know the SumatraPDF project changed the core module to MuPDF.
For MuPDF, I tested it in the following way.
I checked ePub display with mupdf-gl.exe in mupdf-1.21.0-windows.zip, but vertical writing is not supported.

Initially, the Japanese consortium used a special Chrome for ePub verification.
* The special Chrome was the one that used WebKit for the web browser engine.
The display confirmation work has changed in the following order.
1. Special Chrome (using "WebKit" for engine)
2. Old Thorium Reader (using "Readium Desktop" as engine)
3. The latest Thorium Reader (using "Electron 21" as the engine)

Now, you can check vertical writing support by displaying EPUB number 10 on the following page with various web browsers.
https://readium.org/readium-js-viewer/?
The result of vertical writing display is
Microsoft Edge OK
Google Chrome OK
Mozilla Firefox NG

Related Sites
MuPDF
https://www.mupdf.com/
http://git.ghostscript.com/?p=mupdf.git;a=summary
https://github.com/ArtifexSoftware/mupdf
https://mupdf.com/docs/manual-mupdf-gl.html
Webkit
https://webkit.org/
https://github.com/WebKit/WebKit
readium
https://readium.org/
https://github.com/readium
https://github.com/edrlab/thorium-reader
Thorium Reader
https://www.edrlab.org/software/thorium-reader/
https://github.com/edrlab/thorium-reader

**Comment by mark10als** [comment]
The ideal display is shown in the image below.
This is a screenshot of Japanese EPUB software.
![XWS02524](https://user-images.githubusercontent.com/13845687/219281546-68afbdc2-7fb0-4feb-bc69-adaeb7facf27.JPG)
![XWS02525](https://user-images.githubusercontent.com/13845687/219281550-4d6c5ae0-a52e-47be-addb-40f0d8c07c8f.JPG)
![XWS02526](https://user-images.githubusercontent.com/13845687/219281552-f08acbdf-abd1-4b7b-bc98-bb6ae6f55fe9.JPG)

**Comment by GitHubRulesOK** [comment]
The core problem is most digital devices expect a horizontal reading aspect so PDF is perfect for vertical as each character is independently placed however flowing formats are usually RTL or LTR (and those are problematic enough on Japanese built hardware !
![image](https://user-images.githubusercontent.com/12754913/219434187-42677698-c3a3-4934-9223-01aabb4cd112.png)

**Reply by mark10als** [reply]
Thank you for your reply.
Thank you for the comparative test in PDF and EPUB.
I don't think it's easy either.
However, the latest Thorium Reader (using "Electron 21" as the engine) supports vertical writing and right-opening (RTL).
So, is it possible to process PDF and general documents with the MuPDF engine and switch to the Electron engine for EPUB?
The vertical writing support of the WebKit engine is inferior to the Electron engine, so if you want to use it, use the Electron engine.
About page switching.
The current SumatraPDF display is vertical scrolling, so if you want to read EPUB like a book, please support horizontal scrolling!
English books are left bound (LTR), so scrolling starts from the right and disappears to the left.
In Japanese, vertical writing books are right bound (RTL), so scrolling starts from the left and disappears to the right.
However, since Japanese books with horizontal writing are left-bound (LTR), they are scrolled in the same way as English books.
As for right-bound books, I think that the same display method as manga (CBZ) can be used for books with vertical writing in Japanese.
For horizontal writing in English and Japanese, I think that the same display method as American comics (CBZ) may be used.
It does not have to be scrolling.
* Display the cover in the center.
* When creating the atmosphere of a book, display the cover on the left side for manga and display the cover on the right side for American comics.
* It is better to be able to select the display position of the cover.
